core/types/block.go

```diff
@@ -378,6 +428,17 @@ func (b *Block) WithBody(transactions []*Transaction, uncles []*Header) *Block {
 	return block
 }
 
+// WithBody2 returns a new block with the given transaction, uncle, and
+// withdrawal contents.
+func (b *Block) WithBody2(transactions []*Transaction, uncles []*Header, withdrawals []*Withdrawal) *Block {
+	block := b.WithBody(transactions, uncles)
+	if withdrawals != nil {
```

Comment:
This would also capture the case of empty withdrawals array. I noticed some tests failing in the 4844 fork where I'm including this change because of this.
```suggestion
	if withdrawals != nil {
```

Reply:
Oh hmm looks like the spec mandates an empty list if withdrawals are active so scratch that, looks like it's the tests that are the issue.